### PR TITLE
Add core module in Web SDK

### PR DESCRIPTION
**Pull request by jesusbotella**
☝️

### CODE REVIEW

File: src/lib/core/mixins/WithEvents.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import mitt from 'mitt';
+import { CartoError } from '../errors/CartoError';
+
+export abstract class WithEvents {
```
Comment (VictorVelarde) [inline src/lib/core/mixins/WithEvents.ts:4]: ouuu, I see what your're doing here Rick... 👏

**Comment by VictorVelarde** [comment]
Is it ready now? It looks good, but I miss the tests...

And I was just wondering if we should include the namespaced exports on each migration PR

**Comment by jesusbotella** [comment]
I have just added the missing tests from the original module. Everything seems to be working OK.

About the namespacing, I wouldn't export anything besides the credentials from core to the public, and probably we should move that to another folder eventually.

**Review by VictorVelarde — APPROVED** [review]
🆗  Take note of the next refactor steps, and let's move on with this

**Pull request merged**